sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateMutableProjection.scala

```diff
@@ -45,10 +45,32 @@ object GenerateMutableProjection extends CodeGenerator[Seq[Expression], () => Mu
           else
             ${ctx.setColumn("mutableRow", e.dataType, i, evaluationCode.primitive)};
         """
-    }.mkString("\n")
+    }
+
+    val projectionCodeSegments = projectionCodes.grouped(50).toSeq.map(_.mkString("\n"))
+
+    val (projectionCode, projectionFuncs) = if (projectionCodeSegments.length == 1) {
+      (projectionCodeSegments(0), "")
+    } else {
+      val pCode = (0 until projectionCodeSegments.length).map { i =>
+        s"projectSeg$i(_i);"
+      }.mkString("\n")
+
+      val pFuncs = (0 until projectionCodeSegments.length).map { i =>
+        s"""
+          public void projectSeg$i(Object _i) {
```

Comment:
Since the codegen aim to inline the execution, probably we'd better not to increase the overhead for type casting. 
Even, we'd better to put every 50(says) expressions into a single codegen function?